src/WebApiContrib.Core.Versioning/Strategies/IVersionStrategy.cs

```diff
@@ -15,6 +15,6 @@ public interface IVersionStrategy
         /// <param name="context">The <see cref="HttpContext"/> of the current request.</param>
         /// <param name="routeData">The <see cref="RouteData"/>of the current request.</param>
         /// <returns>A version integer, or <c>null</c> if a version cannot be determined.</returns>
-        int? GetVersion(HttpContext context, RouteData routeData);
+        VersionContext GetVersion(HttpContext context, RouteData routeData);
```

Comment:
yeah makes sense. but what would the signature of the interface be? if the result is converted a struct it would need to be nullable struct:

```csharp
VersionResult? GetVersion(HttpContext context, RouteData routeData);
```

because otherwise, if you didn't manage to obtain a version what would you return if not `null`?